tests/net/lib/mqtt_publisher/src/test_mqtt_publish.c

```diff
@@ -254,7 +254,7 @@ static int init_network(void)
 	}
```

Comment:
The patch itself is ok, the commit subject is wrong. Just do "tests: net: mqtt: Fix mqtt_publisher test"

Reply:
Even more, the title line could say what exactly the fix is, e.g. "tests: net: mqtt: Fix rc check in mqtt_publisher test" and second line could be "Actually use the inverted condition."